src/shainet/rnn/rnn.cr

```diff
@@ -0,0 +1,367 @@
+require "log"
+require "./**"
+
+module SHAInet
+  alias RNNLayer = InputLayer | ReluLayer | MaxPoolLayer | FullyConnectedLayer | DropoutLayer | SoftmaxLayer
+  alias RNNPair = {input: Array(Array(Array(Float64))), output: Array(Float64)}
+
+  # Note: Data is stored within specific classes.
+  # Structure hierarchy: RNN > Layer > Filter > Channel > Row > Neuron/Synapse
+
+  # Recurrent Neural Network
+  class RNN
```

Comment:
Why was this file added?

Reply:
My bad, it's a unrelated WIP, it wasn't supposed to be commited in this PR. Sorry about that.